chainindex/events.go

```diff
@@ -0,0 +1,501 @@
+package chainindex
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"math"
+	"sort"
+	"strings"
+
+	"github.com/ipfs/go-cid"
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	amt4 "github.com/filecoin-project/go-amt-ipld/v4"
+	"github.com/filecoin-project/go-state-types/abi"
+	blockadt "github.com/filecoin-project/specs-actors/actors/util/adt"
+
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+type executedMessage struct {
+	msg types.ChainMsg
+	rct types.MessageReceipt
+	// events extracted from receipt
+	evs []types.Event
+}
+
+// events are indexed against their inclusion/message tipset when we get the corresponding execution tipset
+func (si *SqliteIndexer) indexEvents(ctx context.Context, tx *sql.Tx, msgTs *types.TipSet, executionTs *types.TipSet) error {
```

Comment:
add a `if si.idToRobustAddrFunc == nil { error }` near the top here somewhere, since it's possible

Reply:
Done.